### PR TITLE
Pairing & swarming patterns

### PR DESCRIPTION
**Pull request by AdanOrtiz**
Adapt the existing documentation that came out of recent customer engagement to add pairing/swarming patterns to playbook under agile-development/collaboration. Issue #384

### CODE REVIEW

File: agile-development/collaboration/readme.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+**Swarm programming** (“swarming”) - three or more software engineers collaborating on a high-priority item to bring it to completion.
+
+## How to pair program
+
+As mentioned, every story is intentionally assigned to a pair. The pairing assignee may be in the process of upskilling, nevertheless, they are equal partners in the development effort.
+Below are some general guidelines for pairing:
+
+- Upon assignment of the story/product backlog item (PBI), the pair needs to be deliberate about defining how to work together and have a firm definition of the work to be completed. This information should be expressed clearly in the story’s description and acceptance criteria. The expectations about this need to be communicated and agreed upon by both engineers and should be done prior to any actual working sessions.
+- The story owner and pairing assignee do not merely split the work up and sync regularly – they actively work together on the same tasks, often while sharing their screens via a Teams online session.
```
Comment (davidkpiano) [inline agile-development/collaboration/readme.md:20]: Might be good to mention collaborative tools like [VS Live Share](https://visualstudio.microsoft.com/services/live-share/) here, which can be preferable to sharing screens. Not all collaboration needs to be screen-share based.

File: agile-development/collaboration/readme.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+- During the screen-sharing sessions, one engineer provides the development environment while the other actively views and comments verbally.
```
Comment (davidkpiano) [inline agile-development/collaboration/readme.md:21]: Related to previous comment

```suggestion
- During the collaborative sessions, one engineer provides the development environment while the other actively views and comments verbally.
```

File: agile-development/collaboration/readme.md (hunk 1 above)
Comment (AdanOrtiz) [reply agile-development/collaboration/readme.md]: Good point!

File: agile-development/collaboration/readme.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- Swarming allows the collective knowledge and expertise of the team to come together in a focused and unified way.
+- Not only does swarming help close out the item faster, but it also helps the team understand each other’s strengths and weaknesses.
+- Allows the team to build a higher level of trust and work as a cohesive unit.
+
+## When to decide to swarm, pair, and/or split
+
+- While a lot of time can be spent on pair programming, it does make sense to split the work when folks understand how the work will be carried out and the work to be done is largely prescriptive.
+- Once the story has been jointly tasked out by both engineers, the engineers may choose to tackle some tasks separately and then combine the work together at the end.
+- Pair programming is more helpful when the engineers do not have perfect clarity about what is needed to be done or how it can be done.
+- Swarming is done when the two engineers assigned to the story needed an additional sounding board or needed expertise that other team members could provide.
```
Comment (fnocera) [inline agile-development/collaboration/readme.md:67]: NIT: would change "needed" to "need" both times in this sentence.

File: agile-development/collaboration/readme.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+## Benefits of increased collaboration
+
+Knowledge sharing and bringing CSE and customer engineers together in a ‘code-with’ manner is an important aspect of CSE engagement. This grows both our customers’ and our CSE team’s capability to build on Azure. We are responsible for demonstrating engineering fundamentals and leaving the customer in a better place after we disengage. This can only happen if we collaborate and engage together as a team. In addition to improved software quality, this also adds a beneficial social aspect to the engagement.
```
Comment (fnocera) [inline agile-development/collaboration/readme.md:71]: "engagements"

**Review by fnocera — APPROVED** [review]
Thanks for the contribution Adan, this is great to have included in the playbook!

**Review by NarmathaBala — APPROVED** [review]
Looks good, thanks for your contribution!

**Review by davidkpiano — APPROVED** [review]

**Pull request merged**